jobs/cronjob-aws-ocp-snap.yaml

```diff
@@ -0,0 +1,206 @@
+apiVersion: v1
+kind: Template
+metadata:
+  name: aws-ocp-snap
+objects:
+- apiVersion: v1
+  kind: ImageStream
+  metadata:
+    annotations:
+      description: Keeps track of changes in the application image
+    name: ${NAME}
+    labels:
+      template: aws-ocp-snap
+- apiVersion: v1
+  kind: BuildConfig
+  metadata:
+    annotations:
+      description: Defines how to build the application
+    name: ${NAME}
+    labels:
+      template: aws-ocp-snap
+  spec:
+    completionDeadlineSeconds: "1800"
+    output:
+      to:
+        kind: ImageStreamTag
+        name: ${NAME}:latest
+    runPolicy: Serial
+    source:
+      git:
+        uri: https://github.com/redhat-cop/openshift-management.git
+      contextDir: /images/aws-ocp-snap
+    strategy:
+      dockerStrategy:
+        from:
+          kind: DockerImage
+          name: centos:7
+      type: Docker
+    triggers:
+    - type: ConfigChange
+- apiVersion: v1
+  kind: Secret
+  metadata:
+    name: aws-secret
+    labels:
+      template: aws-ocp-snap
+  data:
+    key_id: ${AWS_ACCESS_KEY_ID}
+    access_key: ${AWS_SECRET_ACCESS_KEY}
+    region: ${AWS_REGION}
+- apiVersion: batch/v2alpha1
+  kind: CronJob
+  metadata:
+    name: "${JOB_NAME}"
+    labels:
+      template: aws-ocp-snap
+  spec:
+    schedule: "${SCHEDULE}"
+    concurrencyPolicy: Forbid
+    successfulJobsHistoryLimit: "${SUCCESS_JOBS_HISTORY_LIMIT}"
+    failedJobsHistoryLimit: "${FAILED_JOBS_HISTORY_LIMIT}"
+    jobTemplate:
+      spec:
+        template:
+          spec:
+            containers:
+            - name: "${JOB_NAME}"
+              image: docker-registry.default.svc:5000/${NAMESPACE}/${NAME}:latest
```

Comment:
Could look into new feature in OCP 3.6 which allows native resolution of ImageStreams. Not required, but something to think about

https://docs.openshift.com/container-platform/3.6/dev_guide/managing_images.html#using-is-with-k8s

Reply:
Thanks for the tip @sabre1041, didn't read about this so far, really useful.

Reply:
@sabre1041 yeah, it's a good question re: proper repo ... in some way I believe this could be a better fit for https://github.com/redhat-cop/casl-ansible as that's where we manage and automate the infrastructure side (AWS included). However, this also goes well with the "maintaining your cluster approach" that this current repo is more about. Maybe a topic to be disucss during our weekly call.